kyo-core/shared/src/main/scala/kyo/Clock.scala

```diff
@@ -128,31 +284,242 @@ object Clock:
       *   A new Deadline instance
       */
     def deadline(duration: Duration)(using Frame): Deadline < IO =
-        local.use(_.deadline(duration))
+        use(_.deadline(duration))
+
+    /** Repeatedly executes a task with a fixed delay between completions.
+      *
+      * The delay timer starts after each task completion, making this suitable for tasks that should maintain a minimum gap between
+      * executions.
+      *
+      * @param delay
+      *   The duration to wait after each task completion before starting the next execution
+      * @param f
+      *   The task to execute
+      * @return
+      *   A Fiber that can be used to control or interrupt the recurring task
+      */
+    def repeatWithDelay[E, S](delay: Duration)(f: => Unit < (Async & Abort[E]))(using Frame): Fiber[E, Unit] < IO =
```

Comment:
I feel this bloats clock, but as discussed offline there really isn't a good line between Async and Clock for time based execution. Seems fine.